src/cmake/thirdparty/SetupGeosxThirdParty.cmake

```diff
@@ -194,392 +242,260 @@ if(ENABLE_CALIPER)
-    set(ASMJIT_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/thirdparty/asmjit)
-
-    message( STATUS "setting up MathPresso" )
-    find_path( MATHPRESSO_INCLUDE_DIRS mathpresso/mathpresso.h
-            PATHS  ${MATHPRESSO_DIR}/include
-            NO_DEFAULT_PATH
-            NO_CMAKE_ENVIRONMENT_PATH
-            NO_CMAKE_PATH
-            NO_SYSTEM_ENVIRONMENT_PATH
-            NO_CMAKE_SYSTEM_PATH)
-
-    find_library( MATHPRESSO_LIBRARY NAMES mathpresso
-                PATHS ${MATHPRESSO_DIR}/lib
-                NO_DEFAULT_PATH
-                NO_CMAKE_ENVIRONMENT_PATH
-                NO_CMAKE_PATH
-                NO_SYSTEM_ENVIRONMENT_PATH
-                NO_CMAKE_SYSTEM_PATH)
-
-
-    include(FindPackageHandleStandardArgs)
-    find_package_handle_standard_args(MATHPRESSO  DEFAULT_MSG
-                                    MATHPRESSO_INCLUDE_DIRS
-                                    MATHPRESSO_LIBRARY )
-                                    
-    if (NOT MATHPRESSO_FOUND)
-        message(FATAL_ERROR "MATHPRESSO not found in ${MATHPRESSO_DIR}. Maybe you need to build it")
+else()
+    if(ENABLE_CALIPER)
+        message(WARNING "ENABLE_CALIPER is ON but CALIPER_DIR isn't defined.")
     endif()
 
-    blt_register_library( NAME mathpresso
-                        INCLUDES ${MATHPRESSO_INCLUDE_DIRS}
-                        LIBRARIES ${MATHPRESSO_LIBRARY}
-                        TREAT_INCLUDES_AS_SYSTEM ON )
-    
-    set(MATHPRESSO_FOUND ON CACHE BOOL "")
-    set( thirdPartyLibs ${thirdPartyLibs} mathpresso )
-else()
-    set(MATHPRESSO_FOUND OFF CACHE BOOL "")
-    message(STATUS "Not using mathpresso")
+    set(ENABLE_CALIPER OFF CACHE BOOL "" FORCE)
+    message(STATUS "Not using Caliper.")
 endif()
 
 ################################
-# PUGIXML
+# MATHPRESSO
 ################################
-message( STATUS "setting up pugixml" )
-set(PUGIXML_DIR ${GEOSX_TPL_DIR}/pugixml)
-
+if(DEFINED MATHPRESSO_DIR)
+    message(STATUS "MATHPRESSO_DIR = ${MATHPRESSO_DIR}")
 
-find_path( PUGIXML_INCLUDE_DIRS NAMES pugixml.hpp
-           PATHS  ${PUGIXML_DIR}/include
-           NO_DEFAULT_PATH
-           NO_CMAKE_ENVIRONMENT_PATH
-           NO_CMAKE_PATH
-           NO_SYSTEM_ENVIRONMENT_PATH
-           NO_CMAKE_SYSTEM_PATH)
-
-
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    find_library( PUGIXML_LIBRARY NAMES pugixml
-                  PATHS ${PUGIXML_DIR}/lib
-                  NO_DEFAULT_PATH
-                  NO_CMAKE_ENVIRONMENT_PATH
-                  NO_CMAKE_PATH
-                  NO_SYSTEM_ENVIRONMENT_PATH
-                  NO_CMAKE_SYSTEM_PATH)
+    find_and_register(NAME mathpresso
+                      INCLUDE_DIRECTORIES ${MATHPRESSO_DIR}/include
+                      LIBRARY_DIRECTORIES ${MATHPRESSO_DIR}/lib
+                      HEADER mathpresso/mathpresso.h
+                      LIBRARIES mathpresso)
 
+    set(ENABLE_MATHPRESSO ON CACHE BOOL "" FORCE)
+    set(thirdPartyLibs ${thirdPartyLibs} metis)
```

Comment:
Is this really `metis` and not `mathpresso`?

Reply:
You're right that should be `mathpresso`. I'm not sure what the purpose of `thirdPartyLibs` is though since it's not used anywhere.